### DISCUSSION TITLE
[Contract Verification] Deployed bytecode is not equal to generated one from given source

### DISCUSSION THREAD

**Discussion by Robotvasya** [zkSync-Community-Hub/zksync-developers#69] [Contract Verification] [answered]
### Environment

Testnet

### Issue Type

- [ ] Plugin
- [X] Block Explorer

### Contract Address

0xd932ab3e4768daC5A760DFE38a9F726c7Aec158f

### Compiler Type

Single file

### zkSolc Version

1.3.9

### Solc Version

0.8.19

### Contract Name

HelloWorld

### Contract Code

// SPDX-License-Identifier: MIT
pragma solidity ^0.8.16;
contract HelloWorld {
    function sayHelloWorld() public pure returns (string memory) {
        return "Hello World";
    }
}

### Constructor Arguments

No

### Additional Details

Details are described here:

https://github.com/zksync-sdk/zksync2-python/issues/40

BUT.

The same problem I have if I try to verify deployed smart contract from https://github.com/zksync-sdk/zksync2-examples/blob/main/solidity/storage/build/combined.json. Contract: https://github.com/zksync-sdk/zksync2-examples/blob/main/solidity/storage/Storage.sol file.

**Comment by idea404** [comment, marked as answer]
Please refer to the answer posted on the issue here: https://github.com/zksync-sdk/zksync2-python/issues/40#issuecomment-1655745555

**Comment by GainLee** [comment]
I got the same error when I followed this guide https://era.zksync.io/docs/dev/tutorials/gated-nft-paymaster-tutorial.html#implementing-validateandpayforpaymastertransaction-function on both Windows and WSL

**Comment by tus4ncockverb** [comment]
The issue with verifying contracts on the zkSync testnet has been reported in the zksync-sdk/zksync2-python#40 repository. It also occurs when attempting to verify a contract from the zksync2-examples repository. Further investigation is needed to identify the root cause of this problem and develop a corresponding solution.

**Comment by ljada37** [comment]
The issue seems to be related to verifying a contract deployed using zkSync on testnet, specifically with the zkSync SDK in Python. The problem persists even when attempting to verify contracts deployed from the zksync2-examples repository. Further investigation and potential updates to the verification process or SDK may be needed to address this issue.

**Comment by ljada37** [comment]
It appears that there is an issue with verifying smart contracts deployed using zkSync on the testnet, as reported in the GitHub issue zksync-sdk/zksync2-python#40. This issue persists even when attempting to verify contracts deployed from the zksync2-examples repository. Further investigation and potential updates to the verification process or zkSync SDK may be necessary to resolve this issue.

**Comment by heuresfanegarnold** [comment]
It seems like you are encountering an issue with verifying a contract on zkSync testnet using zkSolc version 1.3.9. This issue has also been reported in the zksync-sdk/zksync2-python repository.

To address this problem:
1. Check if there are any updates or fixes available for the zkSolc plugin that might resolve the verification issue.
2. Ensure that you are providing all the necessary information correctly when attempting to verify the contract, including the correct contract address, compiler versions, and contract code.
3. If possible, try using a different version of zkSolc or zkSync SDK to see if the issue persists.
4. Consider reaching out to the maintainers of the zkSolc plugin or the zkSync community for assistance in resolving the verification problem.

By investigating these steps and possibly seeking help from the community or maintainers, you should be able to address the verification issue and successfully verify your contract on zkSync testnet.

**Comment by boyuanx** [comment]
This is starting to happen again when using zksolc 1.4.0.

**Comment by tumb2ibonsa** [comment]
The issue seems to be related to verifying a contract on zkSync testnet using zkSolc version 1.3.9. You may encounter difficulties with contract verification due to potential compatibility issues between the zkSolc version and the contract code. Consider updating to a newer version of zkSolc or reaching out to the zkSync team for assistance in resolving the verification problem.

**Comment by boyuanx** [comment]
This is fixed with `"@matterlabs/hardhat-zksync-verify": "1.4.1",`

What's with the automated bot replies lol 🥴

**Reply by bxpana** [reply]
You still having an issue with this?

**Comment by camicagochristie** [comment]
Issue: Unable to verify contracts deployed on Testnet using zkSolc version 1.3.9 and Solidity version 0.8.19. Similar verification problem encountered with contracts from zksync2-examples repository.

**Comment by glasrudmary** [comment]
The user is encountering an issue with verifying a Solidity contract on Testnet using zkSolc version 1.3.9. The contract code is provided and there are no constructor arguments. The issue is also reported on the zksync-sdk/zksync2-python repository and the zksync-sdk/zksync2-examples repository. Further investigation and debugging may be required to determine the cause of this issue.
